### DISCUSSION TITLE
How to set AUTO_INCREMENT

### DISCUSSION THREAD

**Discussion by cdoky** [prisma/prisma#22718] [Q&A] [answered]
## Problem
I want set AUTO_INCREMENT start with 1000000

## Suggested solution

```
// like this?
model Order {
    id Int @id @default(autoincrement(100000))
}
```

## Alternatives
```SQL
CREATE TABLE `orders` (
  `order_num` int(11) NOT NULL auto_increment,
) ENGINE=InnoDB AUTO_INCREMENT=1000000 DEFAULT CHARSET=utf8;
```

**Comment by ludralph** [comment]
Hi @cdoky 👋 

Thank you for raising this question.

If you are using Prisma with a PostgreSQL database for example, you can modify the starting value of an autoincrement sequence when you are creating a migration. The process involves manually editing the migration file generated by Prisma because it is not possible to do this within the Schema file.

The steps you would take are outlined below:

1. Define your model in the schema.prisma file and include an `@id` field with `@default(autoincrement())`.

2. Generate a migration with the `prisma migrate dev --create-only` command, which creates a new migration file without applying it.

3. Edit the generated SQL migration file in the `prisma/migrations` folder to include the desired starting value for the autoincrement sequence.

4. Apply the migration with the `prisma migrate dev` command.

For PostgreSQL, the SQL command to alter the sequence might look something like this:
```sql
ALTER SEQUENCE "YourModel_id_seq" RESTART WITH 1000000;
```
Please replace `YourModel_id_seq` with the actual sequence name for your model's id field.

A similar discussion where this was answered can be found [here](https://github.com/prisma/prisma/discussions/18246)


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by cdoky** [reply]
Thank you very much. 
I understand that we can solve this problem by modifying SQL. 
But since we use the ORM framework, can we support this property at the ORM level?

**Reply by ludralph** [reply]
Hi @cdoky, you can create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for having this supported in the ORM.

**Reply by ludralph** [reply]
Hi @cdoky, Did you get a chance to create the feature request and document your use case?

**Reply by cdoky** [reply, marked as answer]
Thank you @ludralph 

I created a request: https://github.com/prisma/prisma/issues/22791
